### PR TITLE
solve bugs in text editing module

### LINKED ISSUE #421: selecting by boundingbox sometimes fails in safari
selecting by bbox only seems to work if the bbox's parent element was selected just before in another selecting mode (safari only)

### PR DESCRIPTION
**Pull request by jacob-hutnyk**
resolves #421 and #413 and implemented drag selecting for bounding boxes

### CODE REVIEW

File: src/utils/Resize.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ function Resize (elementId, neonView, dragHandler) {
       d3.drag()
         .on('start', resizeStart)
         .on('drag', resizeDrag)
-        .on('end', resizeEnd)
+        .on('end', resizeEnd.bind(this))
```
Comment (JRegimbal) [inline src/utils/Resize.js]: Do these changes break resizing staves?

Comment (jacob-hutnyk) [reply src/utils/Resize.js]: "this" only gets called in resizeEnd when calling selectBBox, so the behaviour for resizing staves is the same

File: src/TextEditMode.js [hunk 2]
```diff
@@ -90,17 +87,13 @@ export default class TextEditMode {
...
-            let resize = new Resize(rect.closest('.syl').id, this.neonView, this.dragHandler);
-            resize.drawInitialRect();
-          }
-        });
-      });
+      this.dragHandler = new DragHandler(this.neonView, '.sylTextRect-display');
+      if (this.neonView.NeumeEdit === undefined) {
+        // just in case
+        setSelectHelperObjects(this.neonView, this.dragHandler);
+        clickSelect('#mei_output, #mei_output rect');
```
[the remark this answers is not in the document]
Comment (jacob-hutnyk) [reply src/TextEditMode.js]: fixed in next commit

**Comment by jacob-hutnyk** [comment]
this isn't working in DivaView for some reason. going to do #427 which should make it work in both modes

**Comment by jacob-hutnyk** [comment]
actually it turns out a small hack resolved the problem in diva viewer, meaning #427 is not actually necessary for this pr!

**Pull request merged**